functional-samples/cookbook.offscreen-user-media/manifest.json

```diff
@@ -0,0 +1,25 @@
+{
+  "name": "Offscreen API - User media",
+  "version": "1.0",
+  "manifest_version": 3,
+  "description": "Shows how to record audio in a chrome extension via offscreen document.",
+  "background": {
+    "service_worker": "background.js"
+  },
+  "content_scripts": [
+    {
+      "matches": ["<all_urls>"],
+      "js": ["contentScript.js"]
```

Comment:
you can do just this 
```suggestion
      "js": ["contentScript.js", "requestPermissions.js"]
```
and run it without all the fuss around messaging to offscreen and back

Reply:
however I still get permission in `prompt` state, so I am probably missing something from your code

Reply:
@sorokinvj 

#821  : The reason for requesting mic and cam permissions inside the iframe is because we want to ask for the permissions on behalf of the extension app (`eg -> "[myExtension] wants to Use" instead of "www.google.com wants to use your camera and microphone"`) and hold onto the permissions across all tabs, only asking once for the permissions.

---------------------------------

Your suggestion of injecting CS directly via manifest will not help in accomplishing above goal. But will definitely ease around the fuss if you don't need it in the context of your extension.


Attaching a snapshot for reference how your suggested code will ask for permissions.

![image](https://github.com/GoogleChrome/chrome-extensions-samples/assets/67885496/0cf491ac-6bba-4992-9e83-ccacc1e989d5)

Let me know if I am missing out something in your proposed flow.
Thanks.